### DISCUSSION TITLE
Height Map probing and offsets

### DISCUSSION THREAD

**Discussion by phil-barrett** [terjeio/ioSender#426] [Q&A]
I'm building a small laser machine and a key feature is the ability to map to uneven surfaces.  Height map probing should work but the probe points will need to be offset from the "controlled point" due to the nature of the laser mechanism. The probe must be physically mounted to the side of the laser module.  At first I thought G92 was the way to go but now I am not sure.

For discussion sake, the probe is mounted +5mm X and -5mm Y from the controlled point (CP). If I set G92 X-5 Y5 (which puts the CP over -5,5 and the probe point over 0,0) and then run the height map probing it moves the CP to 0,0 to start probing. This puts the 0,0 probe point over 5,-5. 

What I want is the probing to be offset by -5,5 and then when I apply the height map to the GCode, it uses the map without the offset. In practice, I think the easiest approach for a user is to place the probe over the 0,0 point (with the CP thus positioned over -5,5) and run the height map probing. Then apply the height map with the offset corrected for (ie, starting at 0,0 not -5,5). Also, I want to hide the offset in a macro so people don't need to enter the offset each time (that may be a different issue). I am clearly missing a point (or 2) here.  Any help appreciated.

Phil

**Comment by terjeio** [comment]
So XY offset should be enabled for heightmaps?

![image](https://github.com/user-attachments/assets/c5d92bf8-15cc-4198-819f-84c515273f6e)

>  Also, I want to hide the offset in a macro so people don't need to enter the offset each time

The offset is saved per profile, so switching between offsets is simple - just add/select profile(s).
